### PR TITLE
Add parallax magnification calculation.

### PR DESCRIPTION
**Pull request by NunotaKansuke**
Hello, and thank you for maintaining microlux.
My name is Kansuke Nunota, and I am involved in the MOA and PRIME projects at Osaka University in Japan.

This PR adds support for calculating magnification including parallax effects
by introducing a new `make_binary_mag_with_parallax` function in `model.py`.

### Summary
With this change, parallax magnification can be computed as follows:

```python
binary_mag_par = make_binary_mag_with_parallax(
    ra=ra_deg, dec=dec_deg, tref=tref
)
mag_lux = binary_mag_par(
    t0, u0, tE, rho, q, s, alpha_deg, piEN, piEE, times
)
```

This enables direct evaluation of binary-lens magnification with parallax.

### Background
The parallax implementation was originally developed as part of [the Roman
microlensing anomaly finder in the RGESPIT project](https://github.com/NunotaKansuke/jacscanomaly):

microlux already imports this parallax code, and this PR exposes it for practical
use within microlux.

The implementation is based on ephemeris tables from [JPL Horizons](https://ssd.jpl.nasa.gov/horizons/app.html#/):

### Validation
The accuracy has been validated through comparisons with VBMicrolensing and
MulensModel in the following notebook:
https://github.com/NunotaKansuke/jacscanomaly/blob/main/tests/parallax_test.ipynb

In addition, this PR includes tests using microlux itself:

tests/test_parallax.py

If you would prefer this change to target a different branch or be structured
differently, I would be happy to adjust it.

Thank you for your time and consideration

### CODE REVIEW

**Comment by CoastEgo** [comment]
Hi @NunotaKansuke, thanks for your contributions! It is fantastic to see interest from the MOA/PRIME community. It's indeed essential to add higher-order effects support (e.g., parallax and orbital motion) in`microlux`. 

Actually, I have been thinking along the same lines and implemented a parallax calculation function, similar to `MulensModel`, in my modeling pipeline. I am currently considering a refactoring for `microlux` to support these effects, but I have a few architectural considerations in mind.

1. **Minimizing Dependencies** One major consideration is that we want to minimize external dependencies to reduce maintenance costs and avoid "dependency hell". For example, the latest `jax` requires `Numpy >= 2.0`, while the current version of `MulensModel` doesn't support Numpy 2.0. The current parallax implementation in this PR requires dependencies on `microjax`, `jacscanomaly`, and `jaxopt`, which potentially add maintenance overhead given the frequent updates in the `jax` ecosystem. We prefer a pure JAX-native implementation or relying strictly on highly mature libraries.

2. **Abstract Architecture** Another consideration is that we want to design a general base class, like `Abstract_Trajectory_Model`, to standardize the support for these higher-order effects. A rough example might look like this:
<details>
 <summary>click here</summary>
  
```python

class TrajectoryModel(NamedTuple):
    """
    Parallax trajectory model

    - times: times of the observations in Julian days
    - delta_s: s the projected offset of the Earth-to-Sun vector in AU. see (Gould 2004) for details.
    """

    times: np.ndarray
    delta_s: np.ndarray | None
    photo_center: bool = True

    def calculate_trajectory(self, params: np.ndarray) -> np.ndarray:
        """
        Calculate the microlensing trajectory for the given parameters.

        Args:
            params (np.ndarray): Array containing [t0, u0, te, rho, alpha_rad, s, q, pi_E_N, pi_E_E].
        Returns:
            np.ndarray: Complex trajectory at each time point.
        """
        t0, u0, te, rho, alpha_rad, s, q, pi_E_N, pi_E_E = params

        times = self.times

        vector_tau = (times - t0) / te
        vector_beta = u0 * jnp.ones_like(times)

        pi_E_vector = jnp.array([pi_E_N, pi_E_E])

        if self.delta_s is not None:
            delta_tau_vector = jnp.dot(self.delta_s, pi_E_vector)
            delta_beta_vector = -jnp.cross(self.delta_s, pi_E_vector)

            vector_tau += delta_tau_vector
            vector_beta += delta_beta_vector

        trajectory = vector_tau * jnp.exp(1j * alpha_rad) + 1j * vector_beta * jnp.exp(
            1j * alpha_rad
        )

        if self.photo_center:
            # convert to photocenter coordinates for s>1 case
            masscent_to_magcent = q / (1 + q) * (s - 1 / s)
            cond = s > 1
            shift = jnp.where(cond, masscent_to_magcent, 0)
            trajectory -= shift

        return trajectory
```
</details>

## Next Steps
To align our efforts,  I will open a new PR shortly to introduce the base class. Once that is merged, I would appreciate it if you could open a new PR implementing your parallax logic inherited from this new abstract model.

How does that sound? Thanks again for your time and effort on this!
